### ISSUE TITLE
RPMs don't work on fedora silverblue/kinoite

### ISSUE THREAD

**Issue by aaaa2aaaa** [Frogging-Family/linux-tkg#523]
**Describe the bug**
 - Which OS do you encounter the bug on? Fedora kinoite
 - A clear and concise description of what the bug is. The RPMS produced by linux-tkg cannot be installed by rpm-ostree
 
Normal kernel test:
```
❯ rpm-ostree override replace https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-5.18.0-60.fc37.x86_64.rpm https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-core-5.18.0-60.fc37.x86_64.rpm https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-modules-5.18.0-60.fc37.x86_64.rpm https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-modules-extra-5.18.0-60.fc37.x86_64.rpm 
Downloading https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-5.18.0-60.fc37.x86_64.rpm...done
Downloading https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-core-5.18.0-60.fc37.x86_64.rpm...done
Downloading https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-modules-5.18.0-60.fc37.x86_64.rpm...done
Downloading https://kojipkgs.fedoraproject.org//packages/kernel/5.18.0/60.fc37/x86_64/kernel-modules-extra-5.18.0-60.fc37.x86_64.rpm...done
Checking out tree c048840... done
Enabled rpm-md repositories: fedora-cisco-openh264 fedora-modular updates-modular updates fedora updates-archive
Importing rpm-md... done
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2022-04-07T16:52:38Z solvables: 4
rpm-md repo 'fedora-modular' (cached); generated: 2022-05-04T21:12:01Z solvables: 825
rpm-md repo 'updates-modular' (cached); generated: 2022-05-16T00:18:23Z solvables: 1129
rpm-md repo 'updates' (cached); generated: 2022-05-23T01:03:21Z solvables: 10002
rpm-md repo 'fedora' (cached); generated: 2022-05-04T21:16:11Z solvables: 67992
rpm-md repo 'updates-archive' (cached); generated: 2022-05-23T01:34:06Z solvables: 9491
Resolving dependencies... done
Applying 5 overrides and 12 overlays
Processing packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
Generating initramfs... done
Writing OSTree commit... done
Staging deployment... done
Freed: 1.3 GB (pkgcache branches: 3)
Upgraded:
  kernel 5.17.8-300.fc36 -> 5.18.0-60.fc37
  kernel-core 5.17.8-300.fc36 -> 5.18.0-60.fc37
  kernel-modules 5.17.8-300.fc36 -> 5.18.0-60.fc37
  kernel-modules-extra 5.17.8-300.fc36 -> 5.18.0-60.fc37
Added:
  xsel-1.2.0-30.fc36.x86_64
Use "rpm-ostree override reset" to undo overrides
Run "systemctl reboot" to start a reboot
```
TKG:
```
❯ rpm-ostree override replace kernel-5.17.9_tkg_pds-1.x86_64.rpm kernel-devel-5.17.9_tkg_pds-1.x86_64.rpm kernel-headers-5.17.9_tkg_pds-1.x86_64.rpm
Inactive base replacements:
  kernel-devel-5.17.9_tkg_pds-1.x86_64
  kernel-headers-5.17.9_tkg_pds-1.x86_64
Checking out tree c048840... done
Enabled rpm-md repositories: fedora-cisco-openh264 fedora-modular updates-modular updates fedora updates-archive
Importing rpm-md... done
rpm-md repo 'fedora-cisco-openh264' (cached); generated: 2022-04-07T16:52:38Z solvables: 4
rpm-md repo 'fedora-modular' (cached); generated: 2022-05-04T21:12:01Z solvables: 825
rpm-md repo 'updates-modular' (cached); generated: 2022-05-16T00:18:23Z solvables: 1129
rpm-md repo 'updates' (cached); generated: 2022-05-23T01:03:21Z solvables: 10002
rpm-md repo 'fedora' (cached); generated: 2022-05-04T21:16:11Z solvables: 67992
rpm-md repo 'updates-archive' (cached); generated: 2022-05-23T01:34:06Z solvables: 9491
Resolving dependencies... done
Applying 2 overrides and 12 overlays
Processing packages... done
Running pre scripts... done
Running post scripts... done
Running posttrans scripts... done
Writing rpmdb... done
error: Multiple subdirectories found in: usr/lib/modules: 5.17.8-300.fc36.x86_64 5.17.9_tkg_pds
```

I'm not sure if this is an upstream issue or an issue with the way the RPMs are generated, since it theoretically should be fine:
https://github.com/coreos/rpm-ostree/issues/3530
https://github.com/fedora-silverblue/issue-tracker/issues/214

**Comment by AdelKS** [comment]
Hello, I am note sure if this is related to fixes I implemented recently, please give a try to PR #610 and see if it changes anything.

**Comment by Tk-Glitch** [comment]
Should be fixed with https://github.com/Frogging-Family/linux-tkg/commit/1d063ca32634997616d9cb1da81253bdedc67054
